Math/NumberTheory/Primes/Sieve/SmoothSieve.hs

```diff
@@ -0,0 +1,80 @@
+module Math.NumberTheory.Primes.Sieve.SmoothSieve
+  ( smoothData,
+    smoothNumbers,
+  ) where
+
+import Data.Vector (Vector, replicate, findIndices, accumulate, map)
+import qualified Data.Vector as V
+import qualified Data.Vector.Mutable as MV
+import Control.Monad.ST
+import Control.Monad
+-- Finds the number of primes less than or equal to a given number
+import Math.NumberTheory.Primes.Counting.Impl
+
+smoothData :: Int -> Int -> (V.Vector Int, V.Vector (V.Vector Bool))
```

Comment:
Even while this is a prototype, it is a good habit to use unboxed vectors whenever you see an unboxable type. Most of "small" types are unboxable: `Int`, `Word`, `Bool`, `Double`.

```haskell
import qualified Data.Vector.Unboxed as U 

smoothData :: Int -> Int -> (U.Vector Int, V.Vector (U.Vector Bool))
```

Reply:
Also, do you need a pair of vectors or a vector of pairs is enough? `V.Vector (Int, U.Vector Bool)`.